pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -194,16 +236,52 @@ func buildStatefulSet(mdb mdbv1.MongoDB) (appsv1.StatefulSet, error) {
 		},
 	}
 
+	dataVolume, dataVolumeClaim := buildDataVolumeClaim()
+	v := statefulset.CreateVolumeFromConfigMap("automation-config", "example-mongodb-config")
+	vm := statefulset.CreateVolumeMount("automation-config", "/var/lib/automation/config", "")
+
 	return statefulset.NewBuilder().
 		SetPodTemplateSpec(podSpecTemplate).
 		SetNamespace(mdb.Namespace).
 		SetName(mdb.Name).
 		SetReplicas(mdb.Spec.Members).
 		SetLabels(labels).
 		SetMatchLabels(labels).
+		AddVolumeMount(agentName, vm).
+		AddVolumeMount(mongodbName, vm).
+		AddVolume(v).
+		AddVolumeMount(agentName, dataVolume).
```

Comment:
oh I see, you are using the dataVolume as the shared dir for agent and mongod

I think it's better to use an emptyDir because this is guaranteed to be a local mount, and if the data directory is full, then the agent will still be able to change the config

This looks like a blocker to me

![](https://media.giphy.com/media/10yjJbyWuCN71e/giphy.gif)